presto-main/src/test/java/com/facebook/presto/type/TestIntervalYearMonth.java

```diff
@@ -104,6 +108,11 @@ public void testMultiply()
         assertFunction("2 * INTERVAL '6' YEAR", INTERVAL_YEAR_MONTH, new SqlIntervalYearMonth(12 * 12));
         assertFunction("INTERVAL '1' YEAR * 2.5", INTERVAL_YEAR_MONTH, new SqlIntervalYearMonth((int) (2.5 * 12)));
         assertFunction("2.5 * INTERVAL '1' YEAR", INTERVAL_YEAR_MONTH, new SqlIntervalYearMonth((int) (2.5 * 12)));
+
+        assertNumericOverflow(format("INTERVAL '%s' MONTH * 2", Integer.MAX_VALUE), format("Overflow multiplying interval year-month value by bigint: %s * 2", Integer.MAX_VALUE));
+        assertNumericOverflow(format("2 * INTERVAL '%s' MONTH", Integer.MAX_VALUE), format("Overflow multiplying bigint by interval year-month value: 2 * %s", Integer.MAX_VALUE));
+        assertNumericOverflow(format("INTERVAL '%s' MONTH * 2.0", Integer.MAX_VALUE), format("Overflow multiplying interval year-month value by double: %s * 2.0", Integer.MAX_VALUE));
+        assertNumericOverflow(format("DOUBLE '2' * INTERVAL '%s' MONTH", Integer.MAX_VALUE), format("Overflow multiplying double by interval year-month value: 2.0 * %s", Integer.MAX_VALUE));
```

Comment:
Do you think it's necessary to add a test case like `select INTERVAL '124-30' YEAR TO MONTH * 2200000000`? After this fix we will get an exception with message `Overflow multiplying interval year-month value by double: ...` rather than `Overflow multiplying interval year-month value by bigint: ...`